### PR TITLE
Motor enable pin

### PR DESCRIPTION
**Pull request by dtex**

### CODE REVIEW

File: lib/motor.js [hunk 1]
```diff
@@ -597,7 +599,7 @@ util.inherits(Motor, events.EventEmitter);
 Motor.prototype.initialize = function() {
   this.io.pinMode(this.pins.pwm, this.io.MODES.PWM);
 
-  ["dir", "cdir", "brake"].forEach(function(pin) {
+  ["dir", "cdir", "brake", "enable"].forEach(function(pin) {
     if (this.pins[pin]) {
```
Comment (dtex) [inline lib/motor.js]: Are there any boards where dir, cdir, break or enable could be on pin 0?

Comment (rwaldron) [inline lib/motor.js]: Yeah, actually now that you mention it: Tessel 2.

I'd change this to `if (typeof this.pins[pin] !== "undefined") {`

File: lib/motor.js [hunk 2]
```diff
@@ -727,12 +729,28 @@ Motor.prototype.release = function() {
...
-  this.dir(speed, this.direction);
+  this.dir(this.direction);
   this.start(speed);
 
   return this;
 };
 
+Motor.prototype.enable = function() {
+  var state = priv.get(this);
+  if (this.pins.enable && !this.enabled) {
```
Comment (rwaldron) [inline lib/motor.js]: I guess this would need to change as well: `if (typeof this.pins.enable !== "undefined"...`

File: lib/motor.js [hunk 3]
```diff
@@ -727,12 +729,28 @@ Motor.prototype.release = function() {
...
+    this.setPin(this.pins.enable, 1);
+    state.enabled = true;
+  }
+};
+
+Motor.prototype.disable = function() {
+  var state = priv.get(this);
+  if (this.pins.enable && this.enabled) {
```
Comment (rwaldron) [inline lib/motor.js]: And here

**Comment by rwaldron** [comment]
I'll make an example that shows using the sparkfun block thing.

**Comment by dtex** [comment]
Did you get a chance to test it?

**Comment by rwaldron** [comment]
Not yet, I'm _trying_ to be lazy today, but that's only half working

**Comment by rwaldron** [comment]
@dtex can you update with the changes noted? Thanks!

**Comment by dtex** [comment]
Made that change, but note that I have no way to test in the real world since I don't have that block so hopefully someone can test it out for me.

**Comment by rwaldron** [comment]
I added an example that illustrates the test hardware that I used—thanks again @dtex 

cc @Legendaire

**Comment by Legendaire** [comment]
I fried my motor block so I can't test until the new one arrives.

**Comment by dtex** [comment]
Yikes, how'd that happen?

**Comment by Legendaire** [comment]
Who knew the plus sign meant positive and not the other way around......

**Comment by rwaldron** [comment]
Happens to the best

**Pull request merged**